src/client/bridge/gateway/shard_manager_monitor.rs

```diff
@@ -43,7 +60,7 @@ impl ShardManagerMonitor {
     /// channel (probably indicating that the shard manager should stop anyway)
     ///
     /// [`ShardManagerMessage::ShutdownAll`]: enum.ShardManagerMessage.html#variant.ShutdownAll
-    pub async fn run(&mut self) {
+    pub async fn run(&mut self) -> Result<(), ShardManagerError> {
```

Comment:
The function shall use Serenity's `Result` type alias, `?` should automatically convert the error type appropriately.
```suggestion
    pub async fn run(&mut self) -> Result<()> {
```